src/foam/core/crunch/CapabilityJunctionPayload.js

```diff
@@ -50,7 +50,19 @@ foam.CLASS({
       name: 'status',
       class: 'Enum',
       of: 'foam.core.crunch.CapabilityJunctionStatus',
-      value: foam.core.crunch.CapabilityJunctionStatus.ACTION_REQUIRED
+      value: foam.core.crunch.CapabilityJunctionStatus.ACTION_REQUIRED,
+      javaPostSet: `
+        if ( ((CapabilityJunctionStatus) val) == CapabilityJunctionStatus.GRANTED )
+          setHasBeenGranted(true);
+      `
+    },
+    {
+      name: 'hasBeenGranted',
+      class: 'Boolean',
+      // networkTransient: true,
+      storageTransient: true,
```

Comment:
What happens if we run a script to populate this and request users to provide their occupation, but some of them don’t submit it before the server is restarted?

Reply:
It doesnt need a script, it will autopopulate on restart, then when scripts are run after it will still be valid. The occupation request script should run after restart anyway

Reply:
Adding an occupation un-grants nbpOnboarding, so the status post set action is not executed. How is it then auto-populated?

Reply:
Its auto-populated before the ungranting will happen, right? Since the status of nbpOnboarding wont update until that script is run. So on startup all old users will still have this granted

Reply:
Yes, for the first run.
1. Start the server (hasBeenGranted:true)
2. Un-grant nbpOnboarding (hasBeenGranted:true)
3. Restart server with un-granted nbpOnboarding (hasBeenGranted:false)
So the users who didn't fill occupation before server restart will have hasBeenGranted:false, do they?

Reply:
Ahh, i see, this shouldnt be storageTransient then. Good catch